prboom2/src/gl_main.c

```diff
@@ -2376,31 +2381,17 @@ static void gld_DrawHealthBars(void)
     for (i = count - 1; i >= 0; i--)
     {
       GLHealthBar *hbar = gld_drawinfo.items[GLDIT_HBAR][i].item.hbar;
-      if (hbar->cm != cm)
-      {
-        cm = hbar->cm;
-        glColor4f(cm2RGB[cm][0], cm2RGB[cm][1], cm2RGB[cm][2], 1.0f);
-      }
+      glColor4f(hbar->r, hbar->g, 0, 1);
```

Comment:
Floats should be passed here, not integers, even for fixed values.

Reply:
The compiler would implicitly convert them, especially so for constant expressions. Are you getting a compiler warning or something that would support passing the exact type from the get-go?